### PR TITLE
release lock if unable to sync service correctly.

### PR DESCRIPTION
**Pull request by murphp15**
Please see https://github.com/kube-vip/kube-vip/issues/1032 for details.

### CODE REVIEW

File: pkg/manager/services.go [hunk 1]
```diff
@@ -375,6 +375,11 @@ func (sm *Manager) updateStatus(i *Instance) error {
 			currentServiceCopy.Annotations[requestedIP] = i.dhcpInterfaceIP
 		}
 
+		if currentService.Annotations["development.kube-vip.io/synthetic-api-server-error-on-update"] == "true" {
```
[the remark this answers is not in the document]
Comment (murphp15) [reply pkg/manager/services.go:378]: I added it so that I could test my edge case. 

I would prefer it remain, but also I can see the argument for taking it out? 


What do you mean by write a test around it? do you want unit test or e2e test? 
I was planning on testing it as part of the product we are using this in. Obviously this is in a private repo. 


I'm actually not able to run thee2e tests on a mac with the following error 


```
$(kube_ps1)paulm2@FHC7H7XJ4L kube-vip-2 % make e2e-tests
E2E_IMAGE_PATH=<my-image> go run github.com/onsi/ginkgo/v2/ginkgo --tags=e2e -v -p ./testing/e2e ./testing/e2e/etcd
Failed to compile e2e:

# github.com/insomniacslk/dhcp/dhcpv4/nclient4
../../../../go/pkg/mod/github.com/insomniacslk/dhcp@v0.0.0-20241224095048-b56fa0d5f25d/dhcpv4/nclient4/conn_unix.go:42:59: undefined: unix.ETH_P_IP

Failed to compile etcd:

# github.com/insomniacslk/dhcp/dhcpv4/nclient4
/Users/paulm2/go/pkg/mod/github.com/insomniacslk/dhcp@v0.0.0-20241224095048-b56fa0d5f25d/dhcpv4/nclient4/conn_unix.go:42:59: undefined: unix.ETH_P_IP


Ginkgo ran 2 suites in 848.310541ms

There were failures detected in the following suites:
   e2e ./testing/e2e [Compilation failure]
  etcd ./testing/e2e/etcd [Compilation failure]

Test Suite Failed
exit status 1
```

File: pkg/manager/servicesLeader.go [hunk 2]
```diff
@@ -50,10 +50,12 @@ func (sm *Manager) StartServicesLeaderElection(ctx context.Context, service *v1.
 			Identity: sm.config.NodeName,
 		},
 	}
+	parentCtx, parentCancel := context.WithCancel(ctx)
```
[the remark this answers is not in the document]
Comment (murphp15) [reply pkg/manager/servicesLeader.go]: Done.

**Comment by thebsdbox** [comment]
Looks good, we will leave the annotation in. I can potentially look at writing a services test around it `make services` etc..
